apps/api/src/deployment/services/top-up-custodial-deployments/top-up-custodial-deployments.service.ts

```diff
@@ -41,31 +42,31 @@ export class TopUpCustodialDeploymentsService implements DeploymentsRefiller {
     private readonly drainingDeploymentService: DrainingDeploymentService,
     private readonly rpcClientService: RpcMessageService,
     private readonly blockHttpService: BlockHttpService,
-    private readonly errorService: ErrorService
+    @InjectSentry() private readonly sentry: Sentry,
+    private readonly sentryEventService: SentryEventService
   ) {}
 
   async topUpDeployments(options: TopUpDeploymentsOptions) {
-    const summary = new TopUpSummarizer();
-    summary.set("startBlockHeight", await this.blockHttpService.getCurrentHeight());
+    const summarizer = new TopUpSummarizer();
+    summarizer.set("startBlockHeight", await this.blockHttpService.getCurrentHeight());
 
     const topUpAllCustodialDeployments = this.topUpToolsService.pairs.map(async ({ wallet, client }) => {
       const address = await wallet.getFirstAddress();
       await this.authzHttpService.paginateDepositDeploymentGrants({ grantee: address, limit: this.CONCURRENCY }, async grants => {
-        await Promise.all(
-          grants.map(async grant => {
-            await this.errorService.execWithErrorHandler(
-              { context: TopUpCustodialDeploymentsService.name, grant, event: "TOP_UP_ERROR" },
-              () => this.topUpForGrant(grant, client, options, summary),
-              () => summary.inc("walletsTopUpErrorCount")
-            );
-          })
-        );
+        await Promise.all(grants.map(async grant => this.topUpForGrant(grant, client, options, summarizer)));
       });
     });
     await Promise.all(topUpAllCustodialDeployments);
 
-    summary.set("endBlockHeight", await this.blockHttpService.getCurrentHeight());
-    this.logger.info({ event: "TOP_UP_SUMMARY", summary: summary.summarize() });
+    summarizer.set("endBlockHeight", await this.blockHttpService.getCurrentHeight());
+
+    const summary = summarizer.summarize();
+    this.logger.info({ event: "TOP_UP_SUMMARY", summary, dryRun: options.dryRun });
+
+    if (summary.walletsTopUpErrorCount || summary.deploymentTopUpErrorCount) {
+      const eventId = this.sentry.captureEvent(this.sentryEventService.toEvent({ message: "Top up deployments error", summary }));
+      this.logger.debug({ event: "SENTRY_EVENT_REPORTED", eventId });
```

Comment:
make sure, just one sentry event is reported, the errors could be found via logs